met-web/src/components/publicDashboard/ErrorBox.tsx

```diff
@@ -3,24 +3,27 @@ import { Grid, IconButton, SxProps, Theme } from '@mui/material';
 import { MetBody, MetHeader4, MetPaper } from 'components/common';
 import LoopIcon from '@mui/icons-material/Loop';
 import { DASHBOARD } from './constants';
+import { When } from 'react-if';
```

Comment:
I see you have used the ErrorBox component to display no date, but no data is not an error. It is better if create a noData component and choose accordingly